### PR TITLE
MRG: when lingroups are provided, use them for `csv_summary`

### LINKED ISSUE #3315: `csv_summary` format not created for multiple queries (`tax metagenome`)
When we originally created `csv_summary` output format, we didn't have a way to properly check that summarization happened independently on separate queries, so we limited this output to single-query runs.

That issue has long since been fixed, and `csv_summary` can smoothly output multiple query results, except that we explicitly prohibited it from doing so. 

Fix in progress in #3311

### PR DESCRIPTION
**Pull request by bluegenes**
Currently, when we generate a `csv_summary` with LINs, we get a summary at every single LIN rank, which is a lot of results and not very helpful. LINgroups are our way of linking the LINs (e.g. `14;1;0;0;0;0;0;0;0;0`) to a known name/taxonomic group (e.g. "Phylotype I"). 

This PR changes the behavior of `csv_summary` when a `lingroup` file is provided, limiting summarized reporting to just the named lingroups. While the output is very similar to the `lingroup` output we already have, the most important difference is that the sample name is included in the output, meaning that we get intelligible results when running `tax metagenome` on more than one sample.

Prior `tax metagenome` behavior was to always generate a `lingroup` output file when a `lingroups` file is provided. Here, I disable that for multiple queries, since the results wouldn't make sense. I do not replace it with another default, but  I did add a recommendation to the help + doc. 

In the future, we could consider changing the default `lingroup` output to `csv_summary`, since it's actually useful for multiple files. Or, we could modify the `lingroup` output to include query information.

- Also fixes #3315

### CODE REVIEW

**Comment by bluegenes** [comment]
@sourmash-bio/devs ready for review

File: src/sourmash/tax/tax_utils.py [hunk 1]
```diff
@@ -2073,9 +2076,19 @@ def as_lineage_dict(self, query_info, ranks):
...
+        # if lingroups, convert lingroup number to lingroup name
+        if lingroups is not None and sD["lineage"] in lingroups.keys():
+            sD["lineage"] = lingroups[sD["lineage"]]
+        elif (
+            lingroups
+            and sD["lineage"] != "unclassified"
+            and sD["lineage"] not in lingroups.keys()
+        ):
+            return None
+            # import pdb;pdb.set_trace()
```
Comment (bluegenes) [inline src/sourmash/tax/tax_utils.py]:
```suggestion
```

**Review by ctb — APPROVED** [review]
looks good - esp appreciate the documentation update.

there's some missing code coverage - is this just buggy codecov? I haven't dug in at all.

**Comment by ctb** [comment]
oh! I wanted to suggest that you put the suggested changes to behavior in the PR description into new issues, too; I think they require a major version bump?

**Comment by bluegenes** [comment]
> looks good - esp appreciate the documentation update.
> 
> there's some missing code coverage - is this just buggy codecov? I haven't dug in at all.

looks like it was just buggy codecov!

**Comment by bluegenes** [comment]
> oh! I wanted to suggest that you put the suggested changes to behavior in the PR description into new issues, too; I think they require a major version bump?

now in #3361

**Pull request merged**